environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/instance/enhanced/selectType/WeldInstanceSelectTypeTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.weld.environment.se.test.instance.enhanced.selectType;
+
+import java.lang.reflect.Type;
+
+import org.jboss.arquillian.container.se.api.ClassPath;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.BeanArchive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.weld.environment.se.Weld;
+import org.jboss.weld.environment.se.WeldContainer;
+import org.jboss.weld.inject.WeldInstance;
+import org.jboss.weld.test.util.Utils;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * WeldContainer already gives access to WeldInstance - this tests that you can select based on Type
+ *
+ * @author <a href="mailto:manovotn@redhat.com">Matej Novotny</a>
+ */
+@RunWith(Arquillian.class)
+public class WeldInstanceSelectTypeTest {
+
+    @Deployment
+    public static Archive<?> createTestArchive() {
+        return ClassPath.builder().add(ShrinkWrap.create(BeanArchive.class, Utils.getDeploymentNameAsHash(WeldInstanceSelectTypeTest.class))
+            .addPackage(WeldInstanceSelectTypeTest.class.getPackage())).build();
+    }
+
+    @Test
+    public void testValidSelect() {
+        try (WeldContainer container = new Weld().initialize()) {
+            // Class<T> implements Type, so let's make use of that
+            Type firstType = SomeInterface.class;
```

Comment:
Maybe we should add some tests for parameterized types...

Reply:
Hmm, can add that too. Though there isn't much of a change here (as opposed to WeldEvent where I have such test). You could have done the same before with WeldInstance.